mcs/class/corlib/System.Reflection.Emit/FieldBuilder.cs

```diff
@@ -229,7 +229,12 @@ private void RejectIfCreated ()
 		}
 
 		internal FieldInfo RuntimeResolve () {
-			return typeb.CreateType ().GetField (this);
```

Comment:
I had a different fix for this in the works:

```csharp
return FieldInfo.GetFieldFromHandle (handle, new RuntimeTypeHandle (typeb.CreateType () as RuntimeType));
```

The `MonoClassField*` (nee `RuntimeFieldHandle`) should be the same for the cooked and uncooked type so that should let you get the resolved MonoField object.